src/legacy.rs

```diff
@@ -102,8 +104,8 @@ fn is_rust_hash(s: &str) -> bool {
     s.starts_with('h') && s[1..].chars().all(|c| c.is_digit(16))
 }
 
-impl<'a> fmt::Display for Demangle<'a> {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+impl<'a> Demangle<'a> {
+    pub(crate) fn fmt(&self, f: &mut fmt::Formatter, config: &Config) -> fmt::Result {
```

Comment:
Should probably be named `fmt_with_config`.

Reply:
yeah, sounds like a good idea.

Reply:
fixed.